src/components/PhotoPreview.tsx

```diff
@@ -0,0 +1,107 @@
+"use client";
+
+import { Post } from "@/lib/definitions";
+import { X, Bed, Ruler, Bath, ChevronLeft, ChevronRight } from "lucide-react";
+import Image from "next/image";
+import { Button } from "./ui/button";
+import { Suspense, useEffect, useState } from "react";
+import { Skeleton } from "./ui/skeleton";
+
+type TPreviewProps = {
+  data: Post;
+  handleTogglePreview: () => void;
+};
+export default function PhotoPreview({
+  data,
+  handleTogglePreview,
+}: TPreviewProps) {
+  const [currentIndex, setCurrentIndex] = useState(0);
+
+  const goToPrevious = () => {
+    const isFirstSlide = currentIndex === 0;
+    const newIndex = isFirstSlide ? data?.images.length - 1 : currentIndex - 1;
+    setCurrentIndex(newIndex);
+  };
+
+  const goToNext = () => {
+    const isLastSlide = currentIndex === data?.images.length - 1;
+    const newIndex = isLastSlide ? 0 : currentIndex + 1;
+    setCurrentIndex(newIndex);
+  };
+
+  return (
+    <>
+      <main className="overflow-y-hidden max-h-screen">
+        <div className="absolute inset-0 bg-black/90 z-50 ">
```

Comment:
You can add a backdrop of md to keep the back ground grid image from distracting users

Reply:
Alright boss. Will do that.

Reply:
I will be merging this branch, for me to solve the deployment issue we've been having. You can commit subsequent changes you need to make to this branch.